Update Gem to work with all Solidus versions

kennyadsl opened the pull request:
This PR:

- updates travis config to test against relevant versions of Solidus + add badge in the README
- update rspec config to have feature specs working
- update factory_bot dependency to be able to test against solidus 2.2
- use `SolidusSupport.payment_method_parent_class` so Stripe Gateway works with all Solidus versions

kennyadsl commented on app/models/spree/gateway/stripe_gateway.rb:
```diff
@@ -1,5 +1,5 @@
 module Spree
-  class Gateway::StripeGateway < Gateway
+  class Gateway::StripeGateway < SolidusSupport.payment_method_parent_class(credit_card: true)
```
I don't think it has ever been released or used. It is not listed in the extensions page either. I'd like to also support Solidus 2.2 as well, that's why I used the `payment_method_parent_class` method here.

tvdeyen replied:
That's fine, but the name of the class should be changed IMO. The intent behind solidusio/solidus#2001 was to remove the confusion about what a gateway is. All the "gateway" classes in `solidus_gateway` (the name of the gem itself is completely wrong) are actually payment methods (they inherit from `Spree::PaymentMethod`) and that should be reflected in the name of this class IMO.

kennyadsl replied:
Ah I see what you mean now. I was planning to use this gem into a project where I currently have `solidus_gateway` and I was thinking at a smoother transition. I think that changing the class name would require a migration to update old payments data, right?

tvdeyen replied:
Yes, we would need to run:

```sql
UPDATE spree_payment_methods SET type = 'SolidusStripe::CreditCardPaymentMethod' WHERE type = 'Spree::Gateway::StripeGateway'
```

kennyadsl replied:
I have something in mind to handle this, let me try

kennyadsl commented:
@tvdeyen from your previous review I've fixed specs (actually travis to work with chrome headless) and removed some useless migrations with the last commit, I think we can merge this and try a solution to [this comment](https://github.com/solidusio-contrib/solidus_stripe/pull/2#discussion_r223869134) into a separate PR. Let me know what you do think about it. 🙂

tvdeyen commented on solidus_stripe.gemspec:
```diff
@@ -39,5 +39,6 @@ Gem::Specification.new do |s|
   s.add_development_dependency "capybara"
   s.add_development_dependency "capybara-screenshot"
   s.add_development_dependency "poltergeist", "~> 1.9"
```
Urghs. We should fix this upstream then

kennyadsl replied:
With [this PR](https://github.com/solidusio/solidus_support/pull/10) right?

tvdeyen replied:
Yes
